scripts/west_commands/runners/nrfjprog.py

```diff
@@ -59,50 +60,46 @@ def do_create(cls, cfg, args):
     def ensure_snr(self):
         if not self.snr:
             self.snr = self.get_board_snr()
+        # elif "*" in self.snr:
+        self.snr = self.get_board_snr(self.snr)
 
-    def get_board_snr(self):
+    @staticmethod
+    def verify_snr(snr):
+        if snr == '0':
+            raise RuntimeError('"nrfjprog --ids" returned 0; '
+                                'is a debugger already connected?')
+
+    def get_boards(self):
+        snrs = self.check_output(['nrfjprog', '--ids'])
+        snrs = snrs.decode(sys.getdefaultencoding()).strip().splitlines()
+        if not snrs:
+            raise RuntimeError('"nrfjprog --ids" did not find a board; '
+                               'is the board connected?')
+        return snrs
+
+    def get_board_snr(self, glob = "*"):
         # Use nrfjprog --ids to discover connected boards.
         #
         # If there's exactly one board connected, it's safe to assume
         # the user wants that one. Otherwise, bail unless there are
         # multiple boards and we are connected to a terminal, in which
         # case use print() and input() to ask what the user wants.
 
-        snrs = self.check_output(['nrfjprog', '--ids'])
-        snrs = snrs.decode(sys.getdefaultencoding()).strip().splitlines()
-        if not snrs:
-            raise RuntimeError('"nrfjprog --ids" did not find a board; '
-                               'is the board connected?')
+        snrs = sorted([snr for snr in self.get_boards() if fullmatch(escape(glob).replace(r"\*", ".+"), snr)])
+        if len(snrs) == 0:
+            raise RuntimeError(
+                'There are no boards connected{}.'.format(" matching '{}'".format(glob) if glob != "*" else ""))
         elif len(snrs) == 1:
             board_snr = snrs[0]
-            if board_snr == '0':
-                raise RuntimeError('"nrfjprog --ids" returned 0; '
-                                   'is a debugger already connected?')
+            self.verify_snr(board_snr)
+            print("Using board {}".format(board_snr))
             return board_snr
-        elif not sys.stdin.isatty():
-            raise RuntimeError(
-                f'refusing to guess which of {len(snrs)} '
-                'connected boards to use. (Interactive prompts '
-                'disabled since standard input is not a terminal.) '
-                'Please specify a serial number on the command line.')
```

Comment:
> Some background: I tried
>
>     echo "2" | west flash
>
> but this gives an error
>
>     FATAL ERROR: refusing to guess which of 4 connected boards to use.
> (Interactive prompts disabled since standard input is not a terminal.) Please
> specify a serial number on the command line.
>
> The interactive prompt is not ideal IMO.

I don't understand why this is a reason to remove this code.

The above command seems like an unnecessary use of pipe, with `west flash --snr 2` being the right way to do it.

Can you explain why this is not acceptable?

Reply:
The current patch removes the `input` altogether. I thought it would be better to have a single way to specify the snr, and avoid the extra code.

The current patch also implements globbing the snr instead of allowing the index to the list.